instrumentation/jmx-metrics/javaagent/README.md

```diff
@@ -221,6 +221,67 @@ Thus, the above definitions will create several metrics, named `my.kafka.streams
 
 The metric descriptions will remain undefined, unless they are provided by the queried MBeans.
 
+### State Metrics
+
+Some JMX attributes expose current state as a non-numeric MBean attribute, in order to capture those as metrics it is recommended to use the special `state` metric type.
+For example, with Tomcat connector, the `Catalina:type=Connector,port=*` MBean has `stateName` (of type `String`), we can define the following rule:
+
+```yaml
+---
+rules:
+  - bean: Catalina:type=Connector,port=*
+    mapping:
+      stateName:
+        type: state
+        metric: tomcat.connector
+        metricAttribute:
+          port: param(port)
+          connector_state:
+            ok: STARTED
+            failed: [STOPPED,FAILED]
+            degraded: _
```

Comment:
If we would switch to using an array of objects, for example with `name` and `value`, then we could add a third one with `default` attribute key with value `true|false` to indicate which entry would be the default instead of using a "magic value" to represent this.